### ISSUE TITLE
rpc.mountd uses all the CPU during zfs create

### ISSUE THREAD

**Issue by cmohler** [zfsonlinux/zfs#2750]
HI,
I'm using Ubuntu 14.04lts (kernel 3.13.0-35-generic) and I have installed ZFS on linux from the PPA (not sure how to get the version). 
I have 3724 zfs filesystems one for each users home dir They are shared via "zfs set sharenfs=on"

When I create a new filesystem with "zfs create users/new-name"
NFS will hang and rpc.mountd will be using 100% of the CPU. It will clear in about 5 minutes and allow clients to access the system again. I increased the number of threads for rpc.mountd to 16 and it will peg all 16 threads at 100% cpu usage. 

I have dual xeons 12 cores each and 32gb of ram. 

Any ideas why this happens? Or how to fix it?

Thanks.

**Comment by rubentolosa** [comment]
I'm experiencing the same issue on a Ubuntu Server 14.04 with zfs version 0.6.5.11 and about 280 file systems. 
As I see that cmohler closed this on Aug 25, 2016... does it mean that if I upgrade to version 0.7 I will have it solved?

Thanks!

**Comment by cmohler** [comment]
@rubentolosa  I did exactly what behlendorf said to do. I stopped using zfs to export NFS shares. I added all my NFS shares to /etc/exports. Each filesystem gets its own line. Like this:
/zfs/                      *(rw,no_root_squash,sync,no_subtree_check,fsid=1)
/zfs/user1             *(rw,no_root_squash,sync,no_subtree_check,fsid=2)
...

It's worked great no need to worry about "zfs set sharenfs=on" Just turn it off and don't mess with it anymore. As far as I know upgrading won't fix the issue.
